README.md

```diff
@@ -71,6 +70,24 @@ auth0.clientCredentialsGrant({
 
 Also you can request a token when the user authenticates using any of our client side SDKs, e.g. [auth0.js](https://github.com/auth0/auth0.js).
 
+Or initialize your client class with the ManagementTokenProvider.
+~~~js
+var ManagementClient = require('auth0').ManagementClient;
+var ManagementTokenProvider = require('auth0').ManagementTokenProvider;
+var auth0 = new ManagementClient({
```

Comment:
I'd do a different API like

```js
new ManagementClient({
  domain: "...",
  clientId: "...",
  clientSecret: "...",
});
```

to avoid repeating parameters in the existing one, and it won't clash with the other API with a token since it only expects domain and token.

In the case you need to pass additional CC parameters like scopes or audience you can add them in the root

```js
new ManagementClient({
  domain: "...",
  clientId: "...",
  clientSecret: "...",
  scopes: "read:users write:users read:payment",
  audience: "https://my.app.com/api"
});
```

For specific provider attributes like cache you can use the tokenProvider namespace

```js
{
  tokenProvider: {
    cacheTTLInSeconds: 10,
    // ... etc
  }
}
```

Reply:
I disagree - it just confuses things, and goes against the idea of seperation of concerns, the ManagementClient should just accept a TokenProvider instance rather than magically create one based on certain options being set (i.e. make it explicit - the ManagementClient requires either a token or a tokenProvider options).

if you personally want to avoid repeating parameters the write a `createManagementClientWithTokenProvider()` factory function - that could even be part of the lib.

Reply:
I don't share this concern since node-auth0 is meant mostly for backend usage hence most developers will be using client credentials to get their tokens, and abstracting client credentials will make this library easier to use out of the box.

All that does not imply that we can't share a generic `TokenProvider` as part of this library for other usages

Reply:
IMHO, falling over the "options interface" is somewhat of a bike-shedding exercise. the fact is that some kind of token-provider functionality is pretty fundamental to actually using this whole library in a real-world context (without it a long running application is forced to either restart or manually track "stale-token" errors and recreate the "management client" instance ... neither of which makes for a decent experience)

I'm suggesting that the implementation is technically WAY better than it is now (i.e. non-existent) so blocking the PR on the basis of "we need to abtract the client credentials" is a bad idea - such a refinement is better off being done at a later. 

Put another way, @dctoon has developed what should have been implemented from the get go, in a module that your company released to support/promote you *paid* service. take the PR and sweat the details of the "options interface" at a later date.